docs/concepts.md

```diff
@@ -0,0 +1,265 @@
+# Core Concepts
+
+This page introduces the fundamental concepts behind Wassette and how it bridges the Model Context Protocol (MCP) with WebAssembly Components.
+
+## Model Context Protocol (MCP)
+
+The Model Context Protocol is a standard protocol that defines how AI agents (like Claude, GitHub Copilot, or Cursor) communicate with external tools and services.
+
+### MCP Components
+
+MCP defines several types of components that servers can provide:
+
+#### MCP Servers vs MCP Clients
+
+- **MCP Clients**: The AI agent applications (VS Code, Claude Code, Cursor, Gemini CLI) that connect to MCP servers and use their capabilities
+- **MCP Servers**: Backend services that provide capabilities to AI agents. Wassette is an MCP server.
+
+#### Tools
+
+**Tools** are functions that AI agents can call to perform actions or retrieve information. For example:
+- A weather tool that fetches current weather data
+- A file system tool that reads or writes files
+- A calculation tool that performs mathematical operations
+
+Wassette primarily focuses on **tools** by translating WebAssembly component functions into MCP tools.
+
+#### Prompts (Not Yet Supported)
+
+**Prompts** are reusable templates that help structure conversations with AI agents. These provide standardized ways to interact with certain types of data or workflows.
+
+> **Note**: Wassette does not currently support MCP prompts.
+
+#### Resources (Not Yet Supported)
+
+**Resources** are data sources that can be read by AI agents, such as files, database entries, or API endpoints. Resources allow agents to access contextual information without requiring explicit tool calls.
+
+> **Note**: Wassette does not currently support MCP resources.
+
+## WebAssembly Component Model
+
+WebAssembly (Wasm) is a portable binary instruction format that runs in a sandboxed environment. The **WebAssembly Component Model** extends basic Wasm with standardized interfaces for building composable, reusable components.
+
+### Key Concepts
+
+#### Components
+
+A **component** is a self-contained WebAssembly module with a well-defined interface. Think of components as portable, language-agnostic libraries that can run securely anywhere.
```

Comment:
do not over-explain what wasm components are. Link to this article for further reading: https://www.cncf.io/blog/2024/07/09/webassembly-components-the-next-wave-of-cloud-native-computing/